src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -950,6 +1057,67 @@ private T PutValue<T>(TransactionOperationContext context, string type, Blittabl
             }
         }
 
+        private void PutCertificate(TransactionOperationContext context, string type, BlittableJsonReaderObject cmd, long index, ServerStore serverStore)
+        {
+            try
+            {
+                var certs = context.Transaction.InnerTransaction.OpenTable(CertificatesSchema, CertificatesSlice);
+                var command = (PutCertificateCommand)CommandBase.CreateFrom(cmd);
+
+                using (Slice.From(context.Allocator, command.PublicKeyPinningHash, out Slice hashSlice))
+                using (Slice.From(context.Allocator, command.Name, out Slice keySlice))
+                using (var cert = context.ReadObject(command.ValueToJson(), "inner-val"))
+                {
+                    var existing = serverStore.Cluster.GetCertificateByThumbprint(context, command.Name) ??
+                                  serverStore.Cluster.GetLocalStateByThumbprint(context, command.Name);
```

Comment:
Is this correct? If this is registered locally, and we register it on the cluster, shouldn't we remove it now from the local?

Reply:
We handle this in the caller, DeleteLocalState is called right after PutCertificate